util/src/main/java/tech/pegasys/pantheon/util/enode/EnodeURL.java

```diff
@@ -103,60 +138,23 @@ public static URI asURI(final String url) {
     return fromString(url).toURI();
   }
 
-  private static String getAndValidateNodeId(final Matcher matcher) {
-    final String invalidNodeIdErrorMsg =
-        "Enode URL contains an invalid node ID. Node ID must have 128 characters and shouldn't include the '0x' hex prefix.";
-    final String nodeId = matcher.group("nodeId");
-
-    checkArgument(nodeId.matches(HEX_STRING_PATTERN), invalidNodeIdErrorMsg);
-    checkArgument(nodeId.length() == 128, invalidNodeIdErrorMsg);
-
-    return nodeId;
-  }
-
-  private static InetAddress getAndValidateIp(final Matcher matcher) {
-    final String ipString = matcher.group("ip");
-
-    try {
-      return InetAddresses.forUriString(ipString);
-    } catch (IllegalArgumentException e) {
-      if (e.getMessage().contains("Not a valid URI IP literal: ")) {
-        throw new IllegalArgumentException("Invalid enode URL IP format.");
-      } else {
-        throw e;
-      }
-    }
-  }
-
-  private static Integer getAndValidatePort(final Matcher matcher, final String portName) {
-    int port = Integer.valueOf(matcher.group(portName));
-    checkArgument(
-        NetworkUtility.isValidPort(port),
-        "Invalid " + portName + " port range. Port should be between 0 - 65535");
-    return port;
-  }
-
-  private static OptionalInt getAndValidateDiscoveryPort(final Matcher matcher) {
-    if (matcher.group("discovery") != null) {
-      return OptionalInt.of(getAndValidatePort(matcher, "discovery"));
-    } else {
-      return OptionalInt.empty();
-    }
-  }
-
   public BytesValue getNodeId() {
     return nodeId;
   }
 
-  public String getIp() {
+  public String getIpAsString() {
     return ip.getHostAddress();
   }
 
+  public InetAddress getIp() {
+    return ip;
+  }
+
   public InetAddress getInetAddress() {
     return ip;
   }
```

Comment:
We probably shouldn't have both `getIp` and `getInetAddress` both returning the same thing.

Reply:
Good call - not sure how I missed this since the method is right below the one I added 😂